# Hypothesis to check

In [54]:
# import os
# os.getcwd()
# path = "/project/eLife_retractions"
# os.chdir(path)

- Let's read retractions600.txt and clean the information to produce a dataframe including the DOI  and PubMed_Id of the retracted paper and those of the retracted notes. This dataframe is called df4

In [1]:
import pandas as pd

df =pd.read_csv('retractions600.txt', sep = '\t')

df = df.reset_index()
df

level_0  \
0          600 Items Displayed Out of 791 Item(s) Found   
1     Structure revision of aspergicin by the crysta...   
2     (PHY) Chemistry; (PHY) Crystallography/Spectro...   
3        Natural Product Research ---Taylor and Francis   
4     Department of Chemical Engineering, Foshan Uni...   
5     http://retractionwatch.com/2018/01/16/caught-n...   
6                                                   NaN   
7                                                   NaN   
8     Solar photothermochemical alkane reverse combu...   
9                        (PHY) Chemistry; (PHY) Energy;   
10    Proceedings of the National Academy of Science...   
11    Department of Chemistry and Biochemistry, Univ...   
12    Car safety seats for children: rear facing for...   
13    (B/T) Transportation; (HSC) Medicine - Pediatr...   
14                  Injury Prevention ---BMJ Publishing   
15    Center for Applied Biomechanics, University of...   
16    http://retractionwatch.com/2017/12/15/saving-k...   
17                                                  NaN   
18                                                  NaN   
19    Car safety seats for children: rear facing for...   
20    (B/T) Transportation; (HSC) Medicine - Pediatr...   
21                  Injury Prevention ---BMJ Publishing   
22    Center for Applied Biomechanics, University of...   
23    http://retractionwatch.com/2017/12/15/saving-k...   
24                                                  NaN   
25                                                  NaN   
26    Transgenic mice overexpressing the ALS-linked ...   
27    (BLS) Biology - Cellular; (BLS) Biology - Mole...   
28    Acta Neuropathologica Communications ---Spring...   
29    Center for Translational Research in Neurodege...   
...                                                 ...   
4032                                                NaN   
4033                                                NaN   
4034                                                NaN   
4035                                                NaN   
4036                                                NaN   
4037  Expression of matrix metalloproteinase-26 (MMP...   
4038  (BLS) Anatomy/Physiology; (BLS) Biology - Cell...   
4039                          Life Sciences ---Elsevier   
4040  State Key Laboratory of Reproductive Biology, ...   
4041                                                NaN   
4042                                                NaN   
4043                                                NaN   
4044                                                NaN   
4045  [NEt4][AuCl2][AuCl3]: Solid-State Evidence of ...   
4046  (PHY) Chemistry; (PHY) Crystallography/Spectro...   
4047                         Angewandte Chemie ---Wiley   
4048  School of Chemistry, University of Manchester,...   
4049  Origin of the Obliquities of the Giant Planets...   
4050  (PHY) Space - Astronomy; (PHY) Space - Astroph...   
4051                  Nature ---Nature Publishing Group   
4052  Facultad de Ciencias Astron�micas y Geof�sicas...   
4053  Diffractive electron imaging of nanoparticles ...   
4054          (PHY) Chemistry; (PHY) Materials Science;   
4055  Nature Materials ---Springer - Nature Publishi...   
4056  Department of Physics and Astronomy, Arizona S...   
4057  YRKL sequence of influenza virus M1 functions ...   
4058  (BLS) Biology - Cellular; (BLS) Genetics; (HSC...   
4059  Journal of Virology ---American Society for Mi...   
4060  Department of Microbiology, Immunology and Mol...   
4061                                                NaN   

                                    level_1                 level_2  \
0                                       NaN                     NaN   
1      +Error in Results and/or Conclusions                Feng Zhu   
2                                       NaN              Jie-sen Li   
3                                       NaN             Wu-chen Xie   
4                                 

In [2]:
df2 = df[['level_0','Date/PubMedID/DOI', 'Date/PubMedID/DOI.1']]
df3 = df2.dropna().reset_index(drop = 'True')

#df4 = df3['Date/PubMedID/DOI']
df3


level_0  \
0     Structure revision of aspergicin by the crysta...   
1     (PHY) Chemistry; (PHY) Crystallography/Spectro...   
2        Natural Product Research ---Taylor and Francis   
3     Solar photothermochemical alkane reverse combu...   
4                        (PHY) Chemistry; (PHY) Energy;   
5     Proceedings of the National Academy of Science...   
6     Car safety seats for children: rear facing for...   
7     (B/T) Transportation; (HSC) Medicine - Pediatr...   
8                   Injury Prevention ---BMJ Publishing   
9     Car safety seats for children: rear facing for...   
10    (B/T) Transportation; (HSC) Medicine - Pediatr...   
11                  Injury Prevention ---BMJ Publishing   
12    Transgenic mice overexpressing the ALS-linked ...   
13    (BLS) Biology - Cellular; (BLS) Biology - Mole...   
14    Acta Neuropathologica Communications ---Spring...   
15    Schisandrin B suppresses NLRP3 inflammasome ac...   
16    (BLS) Biochemistry; (BLS) Biology - Cellular; ...   
17    Biochemical and Biophysical Research Communica...   
18    Transition-Metal-Free C-CN/C-H Cross-Coupling:...   
19                                     (PHY) Chemistry;   
20    Organic Letters ---American Chemical Society (...   
21    NAD(P)H:quinone oxidoreductase 1 protects lung...   
22    (BLS) Biochemistry; (BLS) Biology - Cellular; ...   
23          Free Radical Biology & Medicine ---Elsevier   
24    Effects of synthetic magnetic flux in a parity...   
25                                       (PHY) Physics;   
26       Physical Review A ---American Physical Society   
27    Whole-Genome De Novo Sequencing of the Lignin-...   
28    (BLS) Genetics; (BLS) Microbiology; (BLS) Zool...   
29    Genome Announcements ---American Society for M...   
...                                                 ...   
1608  Structure of MsbA from E. coli: a homolog of t...   
1609  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1610  Science ---American Association for the Advanc...   
1611  Structure of the ABC transporter MsbA in compl...   
1612  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1613  Science ---American Association for the Advanc...   
1614  X-ray structure of the EmrE multidrug transpor...   
1615  (BLS) Biology - Molecular; (BLS) Microbiology;...   
1616  Science ---American Association for the Advanc...   
1617  Dual signaling regulated by calcyon, a D1 dopa...   
1618  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1619  Science ---American Association for the Advanc...   
1620  Silencing of B cell receptor signals in human ...   
1621                       (HSC) Medicine - Immunology;   
1622  The Journal of Experimental Medicine ---Rockef...   
1623  Expression of matrix metalloproteinase-26 (MMP...   
1624  (BLS) Anatomy/Physiology; (BLS) Biology - Cell...   
1625                          Life Sciences ---Elsevier   
1626  [NEt4][AuCl2][AuCl3]: Solid-State Evidence of ...   
1627  (PHY) Chemistry; (PHY) Crystallography/Spectro...   
1628                         Angewandte Chemie ---Wiley   
1629  Origin of the Obliquities of the Giant Planets...   
1630  (PHY) Space - Astronomy; (PHY) Space - Astroph...   
1631                  Nature ---Nature Publishing Group   
1632  Diffractive electron imaging of nanoparticles ...   
1633          (PHY) Chemistry; (PHY) Materials Science;   
1634  Nature Materials ---Springer - Nature Publishi...   
1635  YRKL sequence of influenza virus M1 functions ...   
1636  (BLS) Biology - Cellular; (BLS) Genetics; (HSC...   
1637  Journal of Virology ---American Society for Mi...   

                        Date/PubMedID/DOI                  Date/PubMedID/DOI.1  
0                               3/13/2017                             1/8/2018  
1                                28287288                             29310452  
2           10.1080/14786419.2017.1300800        10.1080/14786419.2017.1422331  
3                                3/8/2016                        

In [3]:

a = []
b = []
c = []

DOI_article = []
PubMed_ID_article = []
DOI_retraction = []
PubMed_ID_retraction = []
#Journal = []

for index, row in df3.iterrows():
    #if (index + 1) % 3 == 0:
        #df4.append([row['Date/PubMedID/DOI'],row['Date/PubMedID/DOI.1']])
        #print(row)
    if len(row['Date/PubMedID/DOI']) > 11: # & row['Date/PubMedID/DOI'] != 'unavailable':
        DOI_article.append(row['Date/PubMedID/DOI'])
        DOI_retraction.append(row['Date/PubMedID/DOI.1'])
        #Journal.append(row['level_0'])
        
        PubMed_ID_article.append(df3.iloc[index -1, 1])
        PubMed_ID_retraction.append(df3.iloc[index -1, 2])

        
        #print(row['Date/PubMedID/DOI'],row['Date/PubMedID/DOI.1'])

# df4 = pd.DataFrame({'Journal': Journal,'DOI':DOI_a
df4 = pd.DataFrame({'DOI':DOI_article, 'PubMed_ID_article':PubMed_ID_article, 'DOI_retraction':DOI_retraction, 'PubMed_ID_retraction':PubMed_ID_retraction}).drop_duplicates()

mask3 = df4.DOI_retraction != 'unavailable'
mask4 = df4.PubMed_ID_retraction != "0"
df4 = df4[mask3 & mask4]
df4


DOI                       DOI_retraction  \
0          10.1080/14786419.2017.1300800        10.1080/14786419.2017.1422331   
1                10.1073/pnas.1516945113              10.1073/pnas.1721878115   
2                 10.1136/ip.2006.015115           10.1136/ip.2006.015115eoc1   
4              10.1186/s40478-016-0393-5            10.1186/s40478-017-0502-0   
5             10.1016/j.bbrc.2017.08.079           10.1016/j.bbrc.2017.08.079   
6            10.1021/acs.orglett.7b02473          10.1021/acs.orglett.7b02473   
7    10.1016/j.freeradbiomed.2011.11.027  10.1016/j.freeradbiomed.2017.06.007   
14       10.1001/archpediatrics.2012.999     10.1001/jamapediatrics.2017.3136   
15                    10.1093/ndt/gfm690                   10.1093/ndt/gfx298   
20          10.3109/19401736.2014.987245        10.1080/24701394.2017.1382095   
24                     10.7150/jca.15753                    10.7150/jca.22507   
26                10.1038/sdata.2016.120               10.1038/sdata.2017.119   
28            10.1016/j.gene.2017.07.018           10.1016/j.gene.2017.07.018   
30                    10.1038/ncomms6446                  10.1038/ncomms16134   
31                    10.1093/nar/gks812                   10.1093/nar/gkx657   
33           10.1093/rheumatology/kew027          10.1093/rheumatology/kex259   
36                      10.1038/nbt.3547                 10.1038/nbt0817-797a   
37                    10.1002/cbin.10615                   10.1002/cbin.10615   
39                    10.1002/cbin.10291                   10.1002/cbin.10291   
40            10.1016/j.cell.2014.02.021           10.1016/j.cell.2017.06.044   
42          10.1371/journal.pone.0164378         10.1371/journal.pone.0180272   
43              10.1177/1534735416635275             10.1177/1534735417713534   
44                  10.1021/jacs.6b01652                 10.1021/jacs.7b04407   
46           10.1108/SAJGBR-12-2015-0085            10.1108/SAJBS-06-2017-072   
47          10.1371/journal.pone.0164571         10.1371/journal.pone.0178903   
48               10.1186/1472-6882-14-82            10.1186/s12906-017-1767-3   
49               10.3389/fphy.2017.00009              10.3389/fphy.2017.00015   
50      10.1001/jamapsychiatry.2016.3955     10.1001/jamapsychiatry.2017.0144   
51             10.1017/S1368980016000458            10.1017/S1368980017000623   
52                   10.2147/OTT.S132806                  10.2147/OTT.S140498   
..                                   ...                                  ...   
514            10.1016/j.jep.2005.08.060            10.1016/j.jep.2007.11.012   
515            10.1016/j.jep.2005.01.027            10.1016/j.jep.2007.11.011   
516              10.1126/science.1103776        10.1126/science.320.5873.176a   
517                     10.1038/35102506                  10.1038/nature06819   
518              10.1073/pnas.0403155101              10.1073/pnas.0801065105   
519            10.1016/j.jmb.2006.01.087            10.1016/j.jmb.2008.01.071   
520                  10.1038/nature04836                  10.1038/nature06621   
521              10.2310/6650.2007.00027         10.2310/JIM.0b013e31816c2094   
522              10.1182/blood.V98.1.125         10.1182/blood-2007-09-114066   
523           10.1152/ajpendo.00702.2006   10.1152/ajpendo.zh1-5190-corr.2007   
524             10.1038/sj.emboj.7600343             10.1038/sj.emboj.7601879   
525                    10.1021/bi972113+                    10.1021/bi701501k   
526            10.1016/j.cub.2004.02.003            10.1016/j.cub.2007.09.050   
528              10.1126/science.1071776        10.1126/science.317.5839.748b   
529   10.1212/01.WNL.0000096940.92986.02   10.1212/01.wnl.0000281830.66370.5e   
530         10.1371/journal.pcbi.0030051         10.1371/journal.pcbi.0030158   
531           10.1016/j.aaen.2006.05.005           10.1016/j.aaen.2007.07.002   
532                    10.1021/bi060547b                    10.1021/

- We will look for the text of the papers included in  df in the PubMEd Central Open Access subset. 

- Download files from the PubMed Central Open Access subset.

In [4]:
import os
import sys
from ftplib import FTP
from fnmatch import fnmatch
from urllib.parse import urlparse
from tqdm import tqdm

def download_ftp_file(ftp_client, filename, output_filename):
    # use temp files so we know the file is not complete
    temp_output_filename = output_filename + '.part'
    if os.path.isfile(output_filename):
        os.remove(output_filename)
    with open(temp_output_filename, 'wb') as f:
        # need to change the type to binary for the size command to succeed
        ftp_client.voidcmd('TYPE I')
        total = ftp_client.size(filename)

        with tqdm(total=total, unit='B', unit_scale=True, leave=False, file=sys.stdout) as pbar:
            pbar.set_description(filename)

            def cb(data):
                pbar.update(len(data))
                f.write(data)

            ftp_client.retrbinary('RETR {}'.format(filename), cb)
    os.rename(temp_output_filename, output_filename)

def download_files_from_ftp(ftp_url, file_pattern, output_dir):
    os.makedirs(output_dir, exist_ok=True)
    parsed_ftp_url = urlparse(ftp_url)
    with FTP(parsed_ftp_url.hostname) as ftp_client:
        ftp_client.login()
        ftp_client.cwd(parsed_ftp_url.path)
        filenames = ftp_client.nlst()
        matching_filenames = [fn for fn in filenames if fnmatch(fn, file_pattern)]
        for filename in matching_filenames:
            output_filename = os.path.join(output_dir, filename)
            if os.path.isfile(output_filename):
                print('already downloaded:', output_filename)
            else:
                download_ftp_file(ftp_client, filename, output_filename)
                print('downloaded:', output_filename)

def download_file_from_ftp(ftp_url, output_dir):
    download_files_from_ftp(
        os.path.dirname(ftp_url),
        os.path.basename(ftp_url),
        output_dir
    )

    
data_path = 'data'

# articles
articles_output_dir = os.path.join(data_path, 'FTP_PUBMED_papers')
articles_ftp_url = 'ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_bulk/'
articles_file_pattern = '*.xml.tar.gz'
# Or you could start by just downloading one file
#file_pattern = 'non_comm_use.A-B.xml.tar.gz'

# pmc ids
pmc_ids_output_dir = os.path.join(data_path, 'PMC_ids')
pmc_ids_ftp_url = 'ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/PMC-ids.csv.gz'
data_path = 'data'
pmc_ids_output_dir = os.path.join(data_path, 'PMC_ids')
pmc_ids_ftp_url = 'ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/PMC-ids.csv.gz'
download_file_from_ftp(pmc_ids_ftp_url, pmc_ids_output_dir)


# oa_file_list.csv 
data_path = 'data'
oa_file_list_output_dir = os.path.join(data_path, 'oa_file_list')
oa_file_list_ftp_url = 'ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_file_list.csv'
download_file_from_ftp(oa_file_list_ftp_url, oa_file_list_output_dir)


already downloaded: data/PMC_ids/PMC-ids.csv.gz
already downloaded: data/oa_file_list/oa_file_list.csv


# Let's find the text of the retraction notes included in df4. 

- We will create an object "article.object" for each DOI of the retraction notes.

- NOTE: the following cell can take aseveral hours to run.

In [11]:
import modules.Article_object as Article_object


list_retractions_df4 = []
counter = 0
for doi_retraction in df4["DOI_retraction"]:
    print(counter, "\t", doi_retraction)
    retraction = Article_object.Article_object()
    retraction.doi = doi_retraction
    #retraction.describe()
    retraction.get_pmc_and_journal_from_doi() 
    retraction.get_text_from_tar_file_using_pmc()
    list_retractions_df4.append(retraction)
    counter += 1
    

0 	 10.1080/14786419.2017.1422331
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
1 	 10.1073/pnas.1721878115
                    Journal Title       ISSN      eISSN  Year Volume Issue  \
4814706  Proc Natl Acad Sci U S A  0027-8424  1091-6490  2018    115     3   

         Page                      DOI       PMCID        PMID Manuscript Id  \
4814706  E557  10.1073/pnas.1721878115  PMC5777017  29311292.0           NaN   

        Release Date  
4814706         live  
Checking:  ./data/comm_use.O-Z.xml.tar.gz
Search total time =  0:03:23.616756
Checking:  ./data/non_comm_use.O-Z.xml.tar.gz
Search total time =  0:05:04.197858
2 	 10.1136/ip.2006.015115eoc1
        Journal Title       ISSN      eISSN  Year Volume Issue  \
4800821      Inj Prev  1353-8047  1475-5785  2017    NaN   NaN   

                       Page       

Checking:  ./data/comm_use.O-Z.xml.tar.gz
Article found:  PLoS_One/PMC5446190.nxml
Search total time =  0:01:32.343014
25 	 10.1186/s12906-017-1767-3
                     Journal Title ISSN      eISSN  Year Volume Issue Page  \
4491118  BMC Complement Altern Med  NaN  1472-6882  2017     17   NaN  260   

                               DOI       PMCID        PMID Manuscript Id  \
4491118  10.1186/s12906-017-1767-3  PMC5427531  28494769.0           NaN   

        Release Date  
4491118         live  
Checking:  ./data/comm_use.A-B.xml.tar.gz
Article found:  BMC_Complement_Altern_Med/PMC5427531.nxml
Search total time =  0:00:39.656725
26 	 10.3389/fphy.2017.00015
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
27 	 10.1001/jamapsychiatry.2017.0144
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, 

Article found:  Int_J_Mol_Med/PMC5360352.nxml
Search total time =  0:02:08.484497
49 	 10.1371/journal.pone.0173325
        Journal Title ISSN      eISSN  Year Volume Issue      Page  \
4403676      PLoS One  NaN  1932-6203  2017     12     2  e0173325   

                                  DOI       PMCID        PMID Manuscript Id  \
4403676  10.1371/journal.pone.0173325  PMC5330502  28245281.0           NaN   

        Release Date  
4403676         live  
Checking:  ./data/comm_use.O-Z.xml.tar.gz
Article found:  PLoS_One/PMC5330502.nxml
Search total time =  0:00:54.175298
50 	 10.1186/s13638-016-0799-6
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
51 	 10.1063/1.4975640
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and

Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
75 	 10.1038/ncomms13456
        Journal Title ISSN      eISSN  Year Volume Issue   Page  \
4246139    Nat Commun  NaN  2041-1723  2016      7   NaN  13456   

                         DOI       PMCID        PMID Manuscript Id  \
4246139  10.1038/ncomms13456  PMC5097120  27999431.0           NaN   

        Release Date  
4246139         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Article found:  Nat_Commun/PMC5097120.nxml
Search total time =  0:01:39.758376
76 	 10.1039/C6GC90110G
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
77 	 10.1165/rcmb.555retraction
                     Journal Title       ISS

Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:56.806969
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:04:02.359162
98 	 10.1074/jbc.A110.191379
        Journal Title       ISSN      eISSN  Year Volume Issue   Page  \
4131718   J Biol Chem  0021-9258  1083-351X  2016    291    32  16925   

                             DOI       PMCID        PMID Manuscript Id  \
4131718  10.1074/jbc.A110.191379  PMC4974408  27496966.0           NaN   

        Release Date  
4131718         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.504175
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:59.153404
99 	 10.1074/jbc.A110.183699
        Journal Title       ISSN      eISSN  Year Volume Issue   Page  \
4131717   J Biol Chem  0021-9258  1083-351X  2016    291    32  16924   

                             DOI       PMCID        PMID Manuscript Id  \
4131717  10.1074/jbc.A110.183699  PMC4974407  27496965.0   

        Journal Title       ISSN      eISSN  Year Volume Issue   Page  \
4062771   J Biol Chem  0021-9258  1083-351X  2016    291    21  11462   

                             DOI       PMCID        PMID Manuscript Id  \
4062771  10.1074/jbc.A116.204376  PMC4900289  27226642.0           NaN   

        Release Date  
4062771         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.624830
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:53.045358
118 	 10.1074/jbc.A116.100194
        Journal Title       ISSN      eISSN  Year Volume Issue   Page  \
4062772   J Biol Chem  0021-9258  1083-351X  2016    291    21  11463   

                             DOI       PMCID        PMID Manuscript Id  \
4062772  10.1074/jbc.A116.100194  PMC4900290  27226643.0           NaN   

        Release Date  
4062772         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:52.518128
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Se

Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.315973
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:53.699626
139 	 10.1136/bcr-2012-006529wthd
        Journal Title ISSN      eISSN  Year Volume Issue               Page  \
3907187  BMJ Case Rep  NaN  1757-790X  2016   2016   NaN  bcr2012006529wthd   

                                 DOI       PMCID        PMID Manuscript Id  \
3907187  10.1136/bcr-2012-006529wthd  PMC4736126  26746226.0           NaN   

        Release Date  
3907187         live  
Checking:  ./data/comm_use.A-B.xml.tar.gz
Search total time =  0:01:59.868105
Checking:  ./data/non_comm_use.A-B.xml.tar.gz
Search total time =  0:02:43.203978
140 	 10.1007/s00125-015-3851-x
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
141 	 10.15866/irecos.v11i1.8368
Emp

Checking:  ./data/comm_use.A-B.xml.tar.gz
Search total time =  0:01:55.952268
Checking:  ./data/non_comm_use.A-B.xml.tar.gz
Search total time =  0:02:37.300460
165 	 10.1128/JB.01404-08
        Journal Title       ISSN      eISSN  Year Volume Issue Page  \
1856174   J Bacteriol  0021-9193  1098-5530  2008    191     3  735   

                         DOI       PMCID        PMID Manuscript Id  \
1856174  10.1128/JB.01404-08  PMC2632069  19060155.0           NaN   

        Release Date  
1856174         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.550778
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:53.986299
166 	 10.1073/pnas.1515456112
                    Journal Title       ISSN      eISSN  Year Volume Issue  \
3752619  Proc Natl Acad Sci U S A  0027-8424  1091-6490  2015    112    37   

          Page                      DOI       PMCID        PMID Manuscript Id  \
3752619  E5222  10.1073/pnas.1515456112  PMC4577165  263

Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
188 	 10.1016/j.cell.2015.03.047
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
189 	 10.1007/s11033-015-3873-y
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
190 	 10.1165/rcmb.524retraction
                     Journal Title       ISSN      eISSN  Year Volume Issue  \
4132406  Am J Respir Cell Mol Biol  1044-1549  1535-4989  2015     52     4   

        Page                         DOI       PMCID        PMID  \
4132406  523  10.1165/rcmb.

journal not defined
214 	 10.1155/2015/137312
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
215 	 10.1016/j.fertnstert.2014.12.108
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
216 	 10.1038/ncb3088.
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
217 	 10.1126/science.346.6216.1466-a
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal no

Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.677815
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:54.630016
244 	 10.1042/BJ4610347
        Journal Title       ISSN      eISSN  Year Volume Issue Page  \
4011880     Biochem J  0264-6021  1470-8728  2014    461  Pt 2  347   

                       DOI       PMCID        PMID Manuscript Id Release Date  
4011880  10.1042/BJ4610347  PMC4847221  24966053.0           NaN         live  
Checking:  ./data/comm_use.A-B.xml.tar.gz
Search total time =  0:01:56.405793
Checking:  ./data/non_comm_use.A-B.xml.tar.gz
Article found:  Biochem_J/PMC4847221.nxml
Search total time =  0:02:23.035333
245 	 10.1155/2014/845487
           Journal Title       ISSN      eISSN  Year Volume Issue    Page  \
3270014  Malar Res Treat  2090-8075  2044-4362  2014   2014   NaN  845487   

                         DOI       PMCID        PMID Manuscript Id  \
3270014  10.1155/2014/845487  PMC4082897  25024865.0        

journal not defined
269 	 10.1037/a0035407
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
270 	 10.3892/ijo.2013.2216
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
271 	 10.1021/bi401605t
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
272 	 10.1016/j.cell.2013.10.040
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
273 	 10

journal not defined
293 	 10.1038/nature12164
        Journal Title       ISSN      eISSN  Year Volume Issue Page  \
3563083        Nature  0028-0836  1476-4687  2013    497  7451  652   

                         DOI       PMCID        PMID Manuscript Id  \
3563083  10.1038/nature12164  PMC4383283  23615611.0   NIHMS673050   

        Release Date  
3563083         live  
Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:53.789407
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:54.683965
294 	 10.1371/annotation/255a6a86-fee6-4663-a23f-36e392e87656
        Journal Title ISSN      eISSN  Year Volume Issue  \
2850227      PLoS One  NaN  1932-6203  2013      8     5   

                                                      Page  \
2850227  10.1371/annotation/255a6a86-fee6-4663-a23f-36e...   

                                                       DOI       PMCID  \
2850227  10.1371/annotation/255a6a86-fee6-4663-a23f-36e...  PMC3655198   

       

Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
315 	 https://doi.org/10.1371/annotation/d78ba1b7-79d3-44f0-b632-224fb9149a23
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
316 	 10.2337/db12-wd10
        Journal Title       ISSN      eISSN  Year Volume Issue  Page  \
2645672      Diabetes  0012-1797  1939-327X  2012     61    10  2652   

                       DOI       PMCID        PMID Manuscript Id Release Date  
2645672  10.2337/db12-wd10  PMC3447914  22997433.0           NaN         live  
Checking:  ./data/comm_use.C-H.xml.tar.gz
Search total time =  0:01:54.564137
Checking:  ./data/non_comm_use.C-H.xml.tar.gz
Article found:  Diabetes/PMC3447914.n

Checking:  ./data/comm_use.A-B.xml.tar.gz
Search total time =  0:02:00.588075
Checking:  ./data/non_comm_use.A-B.xml.tar.gz
Search total time =  0:02:45.097286
336 	 10.1016/j.mod.2011.12.001
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
337 	 10.1165/ajrcmb.46.3.414a
                     Journal Title       ISSN      eISSN  Year Volume Issue  \
4132403  Am J Respir Cell Mol Biol  1044-1549  1535-4989  2012     46     3   

        Page                       DOI       PMCID        PMID Manuscript Id  \
4132403  414  10.1165/ajrcmb.46.3.414a  PMC4975103  22383655.0           NaN   

        Release Date  
4132403         live  
Checking:  ./data/comm_use.A-B.xml.tar.gz
Search total time =  0:01:56.828557
Checking:  ./data/non_comm_use.A-B.xml.tar.gz
Search total time =  0:02:38.800971
338 	 10.1098/rsbl.2011.0376
      

journal not defined
359 	 10.1038/ni0811-804a
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
360 	 10.1088/0022-3727/44/29/299801
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
361 	 10.1016/j.tet.2011.05.063
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
362 	 10.1021/ac201144v
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defin

journal not defined
389 	 10.1071/FPv37n12retract
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
390 	 10.1158/0008-5472.CAN-10-3462
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
391 	 10.1158/0008-5472.CAN-10-3463
        Journal Title       ISSN      eISSN  Year Volume Issue  Page  \
3564575    Cancer Res  0008-5472  1538-7445  2010     70    22  9529   

                                   DOI       PMCID        PMID Manuscript Id  \
3564575  10.1158/0008-5472.CAN-10-3463  PMC4384814  21045160.0   NIHMS675117   

        Release Date  
3564575         live  
Checking:  ./data/comm_use.C-H.xml.tar.gz
Search total time =  0:01:58.306982
Checking:  ./dat

Checking:  ./data/comm_use.I-N.xml.tar.gz
Search total time =  0:01:54.528687
Checking:  ./data/non_comm_use.I-N.xml.tar.gz
Search total time =  0:03:55.730595
413 	 10.1016/j.econlet.2010.04.013
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
414 	 10.1063/1.3434333
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
415 	 10.1016/j.jaci.2010.03.020
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
416 	 10.1002/adsc.201090010
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Is

          Journal Title       ISSN      eISSN  Year Volume Issue  Page  \
2577671  Eur J Neurosci  0953-816X  1460-9568  2009     30    11  2235   

                                      DOI       PMCID        PMID  \
2577671  10.1111/j.1460-9568.2009.07066.x  PMC3378390  20128857.0   

        Manuscript Id Release Date  
2577671   NIHMS380200         live  
Checking:  ./data/comm_use.C-H.xml.tar.gz
Search total time =  0:01:56.915163
Checking:  ./data/non_comm_use.C-H.xml.tar.gz
Search total time =  0:02:58.844489
436 	 10.1021/ac9019363
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
437 	 10.1063/1.3247176
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not define

Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
460 	 10.1016/j.cell.2008.06.045
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
461 	 10.1111/j.1365-2958.2008.06351.x
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
462 	 10.1016/j.str.2011.07.010
        Journal Title       ISSN      eISSN  Year Volume Issue  Page  \
2599897     Structure  0969-2126  1878-4186  2011     19     8  1200   

                               DOI       PMCID        PMID Manuscript Id  \
2599897  10.1016/j.str.2011

journal not defined
488 	 10.1104/pp.104.900215
         Journal Title       ISSN      eISSN  Year Volume Issue  Page  \
1114966  Plant Physiol  0032-0889  1532-2548  2007    143     2  1079   

                           DOI       PMCID        PMID Manuscript Id  \
1114966  10.1104/pp.104.900215  PMC1803714  17284584.0           NaN   

        Release Date  
1114966         live  
Checking:  ./data/comm_use.O-Z.xml.tar.gz
Search total time =  0:03:18.246651
Checking:  ./data/non_comm_use.O-Z.xml.tar.gz
Search total time =  0:04:51.493013
489 	 10.1126/science.314.5807.1875b
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_doi did not find pcm and journal
journal not defined
490 	 10.1126/science.314.5807.1875b
Empty DataFrame
Columns: [Journal Title, ISSN, eISSN, Year, Volume, Issue, Page, DOI, PMCID, PMID, Manuscript Id, Release Date]
Index: []
get_pmc_and_journal_from_d

In [20]:
len(list_retractions_df4)

499

In [18]:
df4.shape[0]

499

- Let's save list_retractions_df4 to file

In [16]:
import pickle 

with open('list_retractions_df4_.data', 'wb') as f:
    pickle.dump(list_retractions_df4, f, pickle.HIGHEST_PROTOCOL)
    f.close()


- Read list_retractions_df4 from file

In [19]:
import pickle
with open('list_retractions_df4_.data', 'rb') as f:
    list_retractions_df4 = pickle.load(f)
    f.close()


- Let's do a list of retractions notes (list_Articles_with_text_df4) for which we have found the text

In [21]:
list_Articles_with_text_df4 = []
for article in list_retractions_df4:
    if (article.text != "Article not found" and
        article.text != ""):
        list_Articles_with_text_df4.append(article)
            
    

In [22]:
len(list_Articles_with_text_df4)

60

In [23]:
list_Articles_with_text_df4[2].describe()

pmc: 	 PMC5665028 
pmid: 	  
doi: 	 10.7150/jca.22507 
journal: 	 J Cancer 
date: 	  
text: 	 b'<!DOCTYPE article PUBLIC "-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120' 
references_dois: 	 


- Let's find the pmid and doi of the references included in the retraction notes (list_Articles_with_text_df4)

In [24]:
for article in list_Articles_with_text_df4:
    article.references_dois = article.get_list_references_dois_and_pmids_from_text()
    print(article.references_dois)
    

[]
[]
[]
[]
[{'doi': '', 'pmid': '25399761'}, {'doi': '', 'pmid': '28825418'}]
[]
[{'doi': '', 'pmid': '27736936'}]
[{'doi': '', 'pmid': '27783646'}]
[{'doi': '10.1186/1472-6882-14-82', 'pmid': '24589031'}]
[]
[{'doi': '10.1155/2016/1493098', 'pmid': ''}]
[]
[{'doi': '10.1186/s12889-017-4082-5', 'pmid': '28166756'}]
[{'doi': '', 'pmid': '25807561'}]
[]
[{'doi': '', 'pmid': '24925287'}]
[{'doi': '', 'pmid': '27759640'}]
[]
[]
[{'doi': '', 'pmid': '27897176'}, {'doi': '', 'pmid': '27897171'}]
[]
[]
[{'doi': '', 'pmid': '26817010'}]
[]
[]
[{'doi': '10.1186/1742-2094-9-223', 'pmid': '22999633'}]
[]
[]
[{'doi': '', 'pmid': '26730579'}]
[]
[]
[]
[]
[{'doi': '10.1126/science.1122847', 'pmid': '16690816'}, {'doi': '10.1104/pp.109.147371', 'pmid': '20097794'}]
[]
[]
[{'doi': '10.1186/1471-2393-13-160', 'pmid': ''}]
[]
[{'doi': '', 'pmid': '24069144'}]
[]
[{'doi': '10.1242/bio.20134671', 'pmid': ''}, {'doi': '10.1242/bio.20136775', 'pmid': ''}]
[]
[{'doi': '10.1186/1756-0500-6-182', 'pmid': '236

- Some checkings

In [26]:
list_Articles_with_text_df4[4].describe()

pmc: 	 PMC5566716 
pmid: 	  
doi: 	 10.1038/ncomms16134 
journal: 	 Nat Commun 
date: 	  
text: 	 b'<!DOCTYPE article PUBLIC "-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120' 
references_dois: 	 [{'doi': '', 'pmid': '25399761'}, {'doi': '', 'pmid': '28825418'}]


In [28]:
list_Articles_with_text_df4[4].text 

b'<!DOCTYPE article PUBLIC "-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD v1.0 20120330//EN" "JATS-archivearticle1.dtd">\n<article xmlns:mml="http://www.w3.org/1998/Math/MathML" article-type="retraction"><?properties open_access?><front><journal-meta><journal-id journal-id-type="nlm-ta">Nat Commun</journal-id><journal-id journal-id-type="iso-abbrev">Nat Commun</journal-id><journal-title-group><journal-title>Nature Communications</journal-title></journal-title-group><issn pub-type="epub">2041-1723</issn><publisher><publisher-name>Nature Publishing Group</publisher-name></publisher></journal-meta><article-meta><article-id pub-id-type="pmid">28825417</article-id><article-id pub-id-type="pmc">5566716</article-id><article-id pub-id-type="pii">ncomms16134</article-id><article-id pub-id-type="doi">10.1038/ncomms16134</article-id><article-categories><subj-group subj-group-type="heading"><subject>Retraction</subject></subj-group></article-categories><title-group><article-title

- How many retraction notes included in list_Articles_with_text_df4 have references?

In [30]:
list_Articles_with_text_and_references_df4 = []

for article in list_Articles_with_text_df4:
    if article.references_dois != []:
        list_Articles_with_text_and_references_df4.append(article)
        
len(list_Articles_with_text_and_references_df4)

18

- Some of these references will be the retracted papers themselve and some others will be other articles (that are potentially critizing the retracted paper). 

- Let's create a dataframe with the DOIs and PMIDs of the article retracted, the retraction notes and the references in the retraction notes. 


In [40]:
#Let's find the identifiers of the papers that were retracted:

#df4[df4['doi_retracted'] == article.doi]

import copy

new = 0

retactions_with_new_references = []

#df_ids_retracted_retraction_critizing = pd.DataFrame({})


list_article_retracted_doi = []
list_article_retracted_pmid = []
list_retractions_doi = []
list_retractions_pmid = []
list_references_in_retraction = []

for article in list_Articles_with_text_and_references_df4:
    
    df_ids_original_and_retraction = df4[df4['DOI_retraction'] == article.doi]

    article_retracted_doi = df_ids_original_and_retraction['DOI'].iloc[0]
    article_retracted_pmid = df_ids_original_and_retraction['PubMed_ID_article'].iloc[0]
    list_article_retracted_doi.append(article_retracted_doi)
    list_article_retracted_pmid.append(article_retracted_pmid)
    
    retraction_doi = df_ids_original_and_retraction['DOI_retraction'].iloc[0]
    retraction_pmid = df_ids_original_and_retraction['PubMed_ID_retraction'].iloc[0]
    list_retractions_doi.append(retraction_doi)
    list_retractions_pmid.append(retraction_pmid)

    #print("article retracted: ")
    #print("doi: ", article_retracted_doi)
    #print("pmid: ",  article_retracted_pmid)

    references_ids_in_article = copy.deepcopy(article.references_dois)
    for reference_in_retraction in article.references_dois:
        """
        print(reference_in_retraction['doi'])
        print(df_ids_original_and_retraction['DOI'].iloc[0])
        """
        if (reference_in_retraction['doi'] == article_retracted_doi or
           reference_in_retraction['pmid'] == article_retracted_pmid):
            #print("Reference to original article.")
            references_ids_in_article.remove(reference_in_retraction)
        else:
            #print("NOT the retracted article")                        
            new += 1
            
    list_references_in_retraction.append(references_ids_in_article)
    
    
    #new_entry = pd.merge(df_ids_original_and_retraction, 
    #                       pd.DataFrame({"critizising": [reference_in_retraction],
    #                                    "doi": article_retracted_doi}), 
    #                     on = 'doi')
    
                         # axis = 1,
                         # on="doi",
                         
                        # ignore_index=True)

    #df_ids_retracted_retraction_critizing = df_ids_retracted_retraction_critizing.append(new_entry)
            
    df_ids_retracted_retraction_critizing = pd.DataFrame({"doi" : list_article_retracted_doi,
                                                          "pmid" : list_article_retracted_pmid,
                                                          "doi_retraction" : list_retractions_doi,
                                                          "pmid_retraction" : list_retractions_pmid,
                                                          "ref_in_retraction" : list_references_in_retraction                                                         
                                                         })
        
#dict_references_ids

print("Number of potential articles criticising: ", new)


Number of potential articles criticising:  9


In [41]:
df_ids_retracted_retraction_critizing

doi                  doi_retraction       pmid  \
0             10.1038/ncomms6446             10.1038/ncomms16134   25399761   
1   10.1371/journal.pone.0164378    10.1371/journal.pone.0180272   27736936   
2   10.1371/journal.pone.0164571    10.1371/journal.pone.0178903   27783646   
3        10.1186/1472-6882-14-82       10.1186/s12906-017-1767-3   24589031   
4           10.1155/2017/6258146            10.1155/2017/6258146    6258146   
5      10.1186/s12889-017-4082-5       10.1186/s12889-017-4185-z   28166756   
6   10.1371/journal.pone.0121549    10.1371/journal.pone.0173587   25807561   
7   10.1371/journal.pone.0099630    10.1371/journal.pone.0173325   24925287   
8    10.1097/MD.0000000000005116     10.1097/MD.0000000000006039  10/1/2016   
9             10.1038/ncomms4761             10.1038/ncomms13417   24781844   
10    10.6065/apem.2015.20.4.226      10.6065/apem.2016.21.3.179   26817010   
11       10.1186/1742-2094-9-223       10.1186/s12974-016-0568-9   22999633   
12  10.1371/journal.pone.0146193    10.1371/journal.pone.0151685   26730579   
13       10.3389/fpls.2014.00421         10.3389/fpls.2015.00153   25191336   
14      10.1186/1471-2393-13-160        10.1186/1471-2393-14-202   23937678   
15  10.1371/journal.pone.0072333    10.1371/journal.pone.0095513   24069144   
16         10.1083/jcb.200702053  10.1083/jcb.20070205310112013r   17470638   
17       10.1186/1756-0500-6-182         10.1186/1756-0500-6-332   23648137   

   pmid_retraction                                  ref_in_retraction  
0         28825417                  [{'doi': '', 'pmid': '28825418'}]  
1         28636659                                                 []  
2         28552949                                                 []  
3         28494769                                                 []  
4          1493098      [{'doi': '10.1155/2016/1493098', 'pmid': ''}]  
5         28314383                                                 []  
6         28264014                                                 []  
7         28245281                                                 []  
8         1/1/2017                  [{'doi': '', 'pmid': '27759640'}]  
9         27897154  [{'doi': '', 'pmid': '27897176'}, {'doi': '', ...  
10        27777913                                                 []  
11        27178190                                                 []  
12        26943177                                                 []  
13        25798143  [{'doi': '10.1126/science.1122847', 'pmid': '1...  
14        24993670                                                 []  
15        24728086                                                 []  
16        24165941  [{'doi': '10.1242/bio.20134671', 'pmid': ''}, ...  
17        23962164                                                 []

mask = df_ids_retracted_retraction_critizing["ref_in_retraction"].apply(lambda c: c != [])
df_with_references_in_retraction = df_ids_retracted_retraction_critizing[mask]
df_with_references_in_retraction 

# CONCLUSION 

We got 499 retraction notes from retractionwatch database. Only 60 of these retractions notes were included in the PubMed Open Access subset. Only 18 out of these 60 retraction notes include references. These references can be the retracted paper itself or another paper. Only 9 papers were found to be new references. 

Checking these 9 references we can conclude that the original hypothesis works: those are papers criticising the retracted paper. However, the limited number of new papers precludes the possibility of training a classifier for sentiment analysis.